aten/src/ATen/SparseTensorImpl.cpp

```diff
@@ -54,6 +54,9 @@ IntArrayRef SparseTensorImpl::strides() const {
 bool SparseTensorImpl::is_contiguous(at::MemoryFormat memory_format) const {
   AT_ERROR("sparse tensors do not have is_contiguous");
 }
+bool SparseTensorImpl::is_non_overlapping_and_dense() const {
+  AT_ERROR("sparse tensors do not have is_non_overlapping_and_dense");
```

Comment:
It might be worth adding a version of the "in this context dense means ..." explanation you gave in the issue to the error text here